test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,493 @@
+
+    let token;
+    let owner;
+    let user;
+    let deployer;
+
+    before(async () => {
+        const wallets = await ethers.getSigners();
+        owner = wallets[0];
+        user = wallets[1];
+        deployer = wallets[1];
+
+        hyperliquidInterchainToken = await deployContract(owner, 'HyperliquidInterchainToken', [owner.address]);
+        hyperliquidInterchainTokenDeployer = await deployContract(owner, 'InterchainTokenDeployer', [hyperliquidInterchainToken.address]);
+
+        const salt = getRandomBytes32();
+        const tokenId = getRandomBytes32();
+
+        const tokenAddress = await hyperliquidInterchainTokenDeployer.deployedAddress(salt);
+
+        token = await getContractAt('HyperliquidInterchainToken', tokenAddress, owner);
+
+        await hyperliquidInterchainTokenDeployer
+            .deployInterchainToken(salt, tokenId, owner.address, name, symbol, decimals)
+            .then((tx) => tx.wait);
+
+        await token.mint(owner.address, mintAmount).then((tx) => tx.wait);
+        expect(await token.interchainTokenId()).to.equal(tokenId);
+    });
+
+    describe('HyperliquidInterchainToken', () => {
+        it('Should calculate hardcoded constants correctly', async () => {
+            await expect(deployContract(owner, 'HyperliquidInterchainToken', [owner.address])).to.not.be.reverted;
+        });
+
+        it('should have the interchainTokenDeployer contract as deployer address after initialization', async () => {
+            const deployer = await token.getDeployer();
+            expect(deployer).to.equal(hyperliquidInterchainTokenDeployer.address);
+        });
+
+        it('should get the correct deployer address after updateDeployer', async () => {
+            await token.connect(owner).updateDeployer(deployer.address);
+            const newDeployer = await token.getDeployer();
+            expect(newDeployer).to.equal(deployer.address);
+        });
+
+        it('should revert when non-ITS or non-operator tries to update deployer', async () => {
+            await expect(token.connect(user).updateDeployer(user.address)).to.be.reverted;
+        });
+
+        it('should store deployer in slot 0', async () => {
+            const provider = ethers.provider;
+            const tokenAddress = token.address;
+
+            // Read slot 0 directly
+            const slot0 = await provider.getStorageAt(tokenAddress, 0);
+            const deployerFromSlot0 = '0x' + slot0.slice(-40); // Last 20 bytes
+
+            // Get deployer via contract function
+            const deployerFromContract = await token.getDeployer();
+
+            // Both should match
+            expect(deployerFromSlot0.toLowerCase()).to.equal(deployerFromContract.toLowerCase());
+
+            // Slot 0 should not be empty
+            expect(slot0).to.not.equal('0x0000000000000000000000000000000000000000000000000000000000000000');
+        });
+
+        it('should maintain ERC20 functionality', async () => {
```

Comment:
don't need to re-test token functionality. since the contract is inheriting existing functionality, we can assume it works. If we really need to retest some logic, the same test should be run on the two separate token instances instead of duplicating tests